custom_components/climate_group/climate.py

```diff
@@ -180,33 +181,78 @@ def should_poll(self) -> bool:
     
     async def async_update(self):
         """Query all members and determine the climate group state."""
-        all_states = [self.hass.states.get(x) for x in self._entity_ids]
-        states = list(filter(None, all_states))
-        self._target_temp = _reduce_attribute(states, 'temperature')
-        self._current_temp = _reduce_attribute(states, 'current_temperature')
-        self._min_temp = _reduce_attribute(states, 'min_temp', reduce=max)
-        self._max_temp = _reduce_attribute(states, 'max_temp', reduce=min)
-        self._mode = None
-        all_modes = [state.state for state in states]
-        if all_modes:
-            # Report the most common effect.
-            modes_count = Counter(itertools.chain(all_modes))
-            self._mode = modes_count.most_common(1)[0][0]
+        raw_states = [self.hass.states.get(x) for x in self._entity_ids]
+        all_states = list(filter(None, raw_states))
+
+        states = list(filter(lambda x: x is not None and 'preset_mode' in x.attributes, raw_states))
+
+        heat_states = list(filter(lambda x: x.state == HVAC_MODE_HEAT, states))
+        cool_states = list(filter(lambda x: x.state == HVAC_MODE_HEAT, states))
+
+        heat_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_HEAT, states))
+        cool_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_COOL, states))
+        idle_actions = list(filter(lambda x: x.attributes['hvac_action'] == CURRENT_HVAC_IDLE, states))
+
+        preset_away_stats = list(filter(lambda x: x.attributes['preset_mode'] == PRESET_AWAY, states))
+        preset_none_stats = list(filter(lambda x: x.attributes['preset_mode'] != PRESET_AWAY, states))
+        self._is_away = len(preset_away_stats) == len(states)
```

Comment:
Shouldn't we set the `preset_mode` to the most common preset mode? 

ie:
```
        all_presets = [state.attributes['preset_mode'] for state in states]
        if all_presets:
            # Report the most common preset_mode.
            modes_count = Counter(itertools.chain(all_modes))
            self._preset = modes_count.most_common(1)[0][0]
```

Reply:
This logic allows you to effectively temporarily remove certain thermostats from the group, unless they are all away (e.g. the entire house is empty).

Reply:
Maybe we should not return any preset mode when it is mixed. I think there is also a `UNKNOWN` mode or maybe we can return something like `MULTIPLE`.

Reply:
I thought that was what I did? PRESET_NONE unless all were PRESET_AWAY. That just needs expanding to support other preset modes.

Reply:
fixed - I think, I don't have something that supports more than "away" to test with.